### **Chi- Square Test **

In [11]:
#State The Hypothesis

In [4]:
import numpy as np
import pandas as pd
import scipy.stats as stats

In [8]:
data =  np.array ([[50, 70],
                   [80, 100],
                   [60, 90],
                   [30, 50],
                   [20, 50]])

In [9]:
print(f"Chi-Square Statistic: {chi2}")
print(f"P-value: {p}")
print(f"Degrees of Freedom: {dof}")
print("Expected Frequencies: \n", expected)

Chi-Square Statistic: 5.638227513227513
P-value: 0.22784371130697179
Degrees of Freedom: 4
Expected Frequencies: 
 [[ 48.  72.]
 [ 72. 108.]
 [ 60.  90.]
 [ 32.  48.]
 [ 28.  42.]]


In [10]:
if p < 0.05:
    print("Reject the Null Hypothesis (H₀): There is a significant relationship between product type and satisfaction level.")
else:
    print("Fail to Reject the Null Hypothesis (H₀): There is no significant relationship between product type and satisfaction level.")

Fail to Reject the Null Hypothesis (H₀): There is no significant relationship between product type and satisfaction level.


2. Compute the Chi-Square Statistic:

In [12]:
# Output the Chi-Square Statistic
chi2

5.638227513227513

In [13]:
#3. Determine the Critical Value:

In [14]:
# Calculate the critical value for 4 degrees of freedom at the 0.05 significance level
critical_value = stats.chi2.ppf(1 - 0.05, dof)
critical_value

9.487729036781154


The Chi-Square Statistics is =5.638227513227513


The Critical Value is = 9.487729036781154



**HYPOTHESIS TESTING**

In [26]:
sample_mean = 3050
theoretical_mean = 4000
std_dev = 25
sample_size = 25

In [28]:
standard_error = std_dev / (sample_size ** 0.5)
standard_error

5.0

In [29]:
z_statistic = (sample_mean - theoretical_mean) / standard_error
z_statistic

-190.0

In [30]:
p_value = 2 * (1 - stats.norm.cdf(abs(z_statistic)))
p_value

0.0

In [31]:
print(f"Null Hypothesis (H₀): The mean weekly operating cost is Rs. 4000.")
print(f"Alternative Hypothesis (H₁): The mean weekly operating cost is not Rs. 4000.")
print(f"Z-statistic: {z_statistic}")
print(f"P-value: {p_value}")

Null Hypothesis (H₀): The mean weekly operating cost is Rs. 4000.
Alternative Hypothesis (H₁): The mean weekly operating cost is not Rs. 4000.
Z-statistic: -190.0
P-value: 0.0


In [32]:
if p_value < 0.05:
    print("Reject the Null Hypothesis (H₀): The mean weekly operating cost significantly differs from the theoretical model.")
else:
    print("Fail to Reject the Null Hypothesis (H₀): The mean weekly operating cost does not significantly differ from the theoretical model.")

Reject the Null Hypothesis (H₀): The mean weekly operating cost significantly differs from the theoretical model.


**2. Calculate the Test Statistic:**

In [33]:
sample_mean = 3050
theoretical_mean = 4000
std_dev_units = 5 * 25
sample_size = 25

In [34]:
standard_error = std_dev_units / (sample_size ** 0.5)
standard_error

25.0

In [35]:
test_statistic = (sample_mean - theoretical_mean) / standard_error
test_statistic

-38.0

**3. Determine the Critical Value:**

In [39]:
from scipy.stats import norm

In [37]:
alpha = 0.05

In [40]:
critical_value = norm.ppf(1 - alpha/2)
critical_value

1.959963984540054

In [41]:
print(f"The critical value for an alpha level of {alpha} is: {critical_value:.4f}")

The critical value for an alpha level of 0.05 is: 1.9600


4. Make a Decision:
Compare the test statistic with the critical value to decide whether to reject the null hypothesis

In [42]:
alpha = 0.05

In [43]:
critical_value = norm.ppf(1 - alpha/2)
critical_value

1.959963984540054

In [44]:
test_statistic = -38.0
test_statistic

-38.0

In [45]:
if abs(test_statistic) > critical_value:
    decision = "Reject the null hypothesis"
else:
    decision = "Fail to reject the null hypothesis"

print(f"Critical Value: {critical_value:.4f}")
print(f"Test Statistic: {test_statistic:.4f}")
print(f"Decision: {decision}")

Critical Value: 1.9600
Test Statistic: -38.0000
Decision: Reject the null hypothesis
